src/generic/stage2/numberparsing.h

```diff
@@ -442,8 +425,8 @@ really_inline bool parse_number(UNUSED const uint8_t *const src,
     is_float = true;
     ++p;
     if (!parse_decimal(src, p, i, exponent)) { return false; }
+    digit_count = int(p - start_digits); // used later to guard against overflows
   }
-  int digit_count = int(p - start_digits); // used later to guard against overflows
   if (('e' == *p) || ('E' == *p)) {
```

Comment:
Could it be better when replaced by `'e' == *p | 0x20` ?

Reply:
Probably not. LLVM clang already does this optimization. GCC and VS do something else, but they all bring it down to the same instruction count. No compiler does two comparison.